### PR TITLE
Payroll email

### PR DESCRIPTION
**Pull request by benghaziboy**
@MichaelButkovic @mparent61

### CODE REVIEW

**Comment by benghaziboy** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;9fZV93da&#x2F;263-freelancer-email-0-total"> Freelancer Email: $0 total</a>

**Comment by benghaziboy** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;rcxoO5Ee&#x2F;265-freelancer-email-staff-vs-freelancer"> Freelancer Email: Staff vs. Freelancer</a>

**Comment by benghaziboy** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;SepUKXzU&#x2F;270-freelancer-report-time-zone-issue"> Freelancer Report: time zone issue</a>

**Comment by benghaziboy** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;uyM3e7oO&#x2F;266-freelancer-email-copy-changes"> Freelancer Email: copy changes</a>

**Comment by benghaziboy** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;czvneBCa&#x2F;269-freelancer-email-feature-type-showing-up-as-article"> Freelancer Email: Feature type showing up as &#39;article&#39;</a>

**Comment by benghaziboy** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;l1gBotvB&#x2F;268-freelancer-email-export-data-in-line-items"> Freelancer Email: Export data in &#x2F;line-items</a>

**Comment by benghaziboy** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;aBUNVz0y&#x2F;267-freelancer-report-line-items"> Freelancer Report: Line Items</a>

**Comment by benghaziboy** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;XfyuQME1&#x2F;264-0-5-freelancer-email-deleting-bylines"> (0.5) Freelancer Email: Deleting Bylines</a>

File: bulbs/contributions/filters.py [hunk 1]
```diff
@@ -42,6 +37,21 @@ def apply_published_filter(self, queryset, operation, value):
...
+
+    def get_end(self, request):
+        end_value = self.get_date_datetime_param(request, "end")
+        if end_value:
+            end_value += timezone.timedelta(days=1)
+            end_value -= timezone.timedelta(seconds=1)
+            try:
+                return timezone.make_aware(end_value).astimezone(timezone.pytz.utc)
+            except:
+                import pdb; pdb.set_trace()
```
Comment (MichaelButkovic) [inline bulbs/contributions/filters.py]: do you want to throw an error or something here instead?

**Comment by MichaelButkovic** [comment]
otherwise, LGTM

File: bulbs/contributions/filters.py (hunk 1 above)
Comment (benghaziboy) [inline bulbs/contributions/filters.py]: ah whoops that just shouldn't be here.

File: bulbs/contributions/email.py [hunk 2]
```diff
@@ -16,6 +19,117 @@
...
+    @property
+    def deadline(self):
+        """Return next day as deadline if no deadline provided."""
+        if not self._deadline:
+            self._deadline = self.now + timezone.timedelta(days=1)
+        return self._deadline
+
+    @property
+    def start(self):
+        if not self._start:
```
Comment (mparent61) [inline bulbs/contributions/email.py]: we should make/grab a "memoize" decorator for this stuff in the future

**Comment by mparent61** [comment]
👍

**Comment by benghaziboy** [comment]
@mparent61 @MichaelButkovic updated some stuff

File: bulbs/contributions/templates/reporting/__byline_report.html [hunk 3]
```diff
@@ -0,0 +1,119 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml">
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <title>A Simple Responsive HTML Email</title>
+    <style type="text/css">
```
Comment (mparent61) [inline bulbs/contributions/templates/reporting/__byline_report.html]: you know how to style???

File: bulbs/contributions/views.py [hunk 4]
```diff
@@ -151,6 +152,38 @@ def get_queryset(self):
         return qs
 
 
+class LineItemReportingViewSet(viewsets.GenericViewSet, mixins.ListModelMixin):
+
+    renderer_classes = (
+        LineItemRenderer,
+    ) + tuple(
```
Comment (mparent61) [inline bulbs/contributions/views.py]: I like this `tuple` use instead of just parens with an easy-to-forget trailing comma

**Comment by mparent61** [comment]
LGTM

File: bulbs/contributions/templates/reporting/__byline_report.html (hunk 3 above)
Comment (benghaziboy) [inline bulbs/contributions/templates/reporting/__byline_report.html]: nope. this is what we in the biz call "aping"

**Comment by MichaelButkovic** [comment]
👍

**Pull request merged**